domehunter/dome_control.py

```diff
@@ -0,0 +1,674 @@
+"""Run dome control on a raspberry pi GPIO."""
+
+
+import sys
+import time
+import warnings
+
+from gpiozero import Device, DigitalInputDevice, DigitalOutputDevice
+from gpiozero.pins.mock import MockFactory
+
+from ._astropy_init import *
+
+# if we want to use the automation hat status lights we need to
+# import the pimoroni led driver
+try:  # pragma: no cover
+    import sn3218
+    sn3218.disable()
+except OSError:  # pragma: no cover
+    warnings.warn(
+        "AutomationHAT hardware not detected, testing=True and"
+        " debug_lights=False recommended.")
+except:  # pragma: no cover
+    warnings.warn(
+        "Something went wrong in importing sn3218,"
+        " status lights unlikely to work.")
+
+
+# ----------------------------------------------------------------------------
+
+
+# Enforce Python version check during package import.
+# This is the same check as the one at the top of setup.py
+__minimum_python_version__ = "3.6"
+
+
+class UnsupportedPythonError(Exception):  # pragma: no cover
+    pass
+
+
+minimum = tuple((int(val) for val in __minimum_python_version__.split('.')))
+if sys.version_info < minimum:  # pragma: no cover
+    raise UnsupportedPythonError(
+        f'domehunter does not support Python < {__minimum_python_version__}')
+
+if not _ASTROPY_SETUP_:
+    # For egg_info test builds to pass, put package imports here.
+    pass
+
+
+# TODO: set up functions to raise custom exceptions to allow server to
+#       properly handle exceptions
+class DomeCommandError(Exception):  # pragma: no cover
+    # generic dome command error, placeholder for now
+    pass
+
+
+class Dome():
```

Comment:
```suggestion
class Dome(object):
```

Good habit to get into (for some reason).

Reply:
I thought the only reason to explicitly inherit from `object` was Python 2 compatibility, because in Python 2 that is used to distinguish between 'old style' and 'new style' classes. In Python 3 it's redundant because classes are always 'new style' and always inherit from 'object'.